### PR TITLE
add: stripe tax to onboarding tax task

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

3 possible screens:

- Manual setup: no tax plugins recommended (e.g Afghanistan)
- WooCommerce Tax & Stripe Tax recommended (e.g UK, US)
- Stripe Tax recommended (e.g Singapore)

1. Setup WooCommerce with a country to show one of the above flows
2. All flows should work as expected. (Stripe Tax is the newly added one but the components for all of them saw some change)
3. Stripe tax flow should install the plugin, then bring you to WC Settings
4. WooCommerce tax flow should go through the WC Tax automated setup flow

image
image
image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by ilyasfoo** [comment]
Hey, @rjchow! Can you provide the internal reference for this work?

File: plugins/woocommerce/package.json [hunk 1]
```diff
@@ -688,7 +688,6 @@
 				"node_modules/@woocommerce/e2e-core-tests/CHANGELOG.md",
 				"node_modules/@woocommerce/api/dist/",
 				"node_modules/@woocommerce/admin-e2e-tests/build",
-				"node_modules/@woocommerce/classic-assets/build",
```
Comment (ilyasfoo) [inline plugins/woocommerce/package.json]: Is this removal intended?

Comment (chihsuan) [inline plugins/woocommerce/package.json]: I think we still need to include the `node_modules/@woocommerce/classic-assets/build` here.

I was also encountering the similar thing, I found that if you run `pnpm install` in sub directory of the project, it may update `dependencyOutputs` in wrong way. If you run `pnpm install` in the root directory of the project, it will not change `dependencyOutputs`.

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php [hunk 2]
```diff
@@ -162,4 +164,54 @@ public static function get_automated_support_countries() {
 
 		return $tax_supported_countries;
 	}
+
+	/**
+	 * Get an array of countries that support Stripe tax.
+	 *
+	 * @return array
+	 */
+	public static function get_stripe_tax_support_countries() {
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php]: What do you think if we change the function to be private so that we can remove it when we don't promote the Stripe tax?

File: plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		<PartnerCard
+			name={ __( 'Stripe Tax', 'woocommerce' ) }
+			logo={ StripeTaxLogo }
+			description={ __( 'Powerful global tax tool', 'woocommerce' ) }
+			benefits={ [
+				__( 'Real-time sales tax calculation', 'woocommerce' ),
+				__( 'Multi-economic nexus compliance', 'woocommerce' ),
+				__( 'Detailed tax transaction reports', 'woocommerce' ),
+				__( 'Coverage in over 55 countries', 'woocommerce' ),
+				__( 'Free to install, then pay as you go.', 'woocommerce' ),
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx]: Only this one has a period at the end. The others don't. Should we keep it consistent?

File: plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx [hunk 4]
```diff
@@ -174,39 +177,43 @@ export const Tax: React.FC< TaxProps > = ( { onComplete, query, task } ) => {
...
+	useEffect( () => {
 		if ( query.partner ) {
 			return;
 		}
-
 		recordEvent( 'tasklist_tax_view_options', {
 			options: partners.map( ( partner ) => partner.id ),
 		} );
-	}, [ onAutomate, partners, query ] );
+	}, [ partners, query ] );
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx]:
```suggestion
	}, [ partners, query.partner ] );
```

Should we use `query.partner` instead to make it more specific? Otherwise this may either:

- Trigger when unrelevant properties change
- or doesn't trigger if `partner` is updated via mutation

**Review by chihsuan — APPROVED** [review]
Tested well. 👍  Just left some minor comments.

Also, I noticed that when using JN site, after installing the Stripe Tax, I was redirected to the Main Setting page, not to the Tax settings page and it's due to folder name again. 😅  Looks like Strip Tax checks WC folder to decide weather WC is installed or not. I think we should suggest them using other method to check WC installation. This causes confusion for users.


Screenshot 2024-08-28 at 11 32 40

**Comment by chihsuan** [comment]
> Hey, @rjchow! Can you provide the internal reference for this work?

@ilyasfoo I found it here: p7fD6U-eHc-p2#comment-28712

File: plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx (hunk 3 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx]: Nice catch!

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php (hunk 2 above)
Comment (rjchow) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php]: Sure thing 😄

File: plugins/woocommerce/package.json (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce/package.json]: Aaaaah I usually remove it from the commit but I missed it this time. Thanks for the tip to run it in the root dir! I was wondering what I was doing wrong

File: plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx (hunk 3 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx]: I suspect that was not part of the bulleted list now that I look at it - I think it might have been part of the "Terms". I've left stakeholders a screenshot to confirm

File: plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx [hunk 5]
```diff
@@ -174,39 +177,43 @@ export const Tax: React.FC< TaxProps > = ( { onComplete, query, task } ) => {
...
+	const currentPartner = useMemo( () => {
 		if ( ! query.partner ) {
 			return null;
 		}
 
 		return (
 			partners.find( ( partner ) => partner.id === query.partner ) || null
 		);
-	};
+	}, [ partners, query ] );
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx]:
```suggestion
	}, [ partners, query.partner ] );
```

Same suggestion as above

File: plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+			) : (
+				<Plugins
+					installText={ __( 'Install for free', 'woocommerce' ) }
+					onClick={ () => {
+						recordEvent( 'tasklist_tax_select_option', {
+							selected_option: STRIPE_TAX_PLUGIN_SLUG,
+						} );
+					} }
+					onComplete={ () => {
+						recordEvent( 'tasklist_tax_installed_plugin', {
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx]: I'm curious if this is a new track? I can't find any reference to similar components which fires this track.

If it's new, I'd suggest using `tasklist_tax_install_plugin_success` to align it with error track

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php [hunk 7]
```diff
@@ -162,4 +164,54 @@ public static function get_automated_support_countries() {
...
+
+	/**
+	 * Get an array of countries that support Stripe tax.
+	 *
+	 * @return array
+	 */
+	private static function get_stripe_tax_support_countries() {
+		// https://docs.stripe.com/tax/supported-countries#supported-countries accurate as of 2024-08-26.
+		// countries with remote sales not included.
+		return array(
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php:176]: From the [docs](https://docs.stripe.com/tax/supported-countries#supported-countries), it seems Ukraine `UA` is missing in our list. However, it's also missing from the screenshot in p7fD6U-eHc-p2#comment-28712.

Should we include Ukraine?

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Thanks for the work, @rjchow!

I have a few comments and this one can probably be a follow-up:

I think we also need to consider amending the tax task completion to support whatever Stripe tax is doing, as how [we did with WCT](https://github.com/woocommerce/woocommerce/blob/4270fa07e6dd035b326baae736cfe611ac819611/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php#L112-L113). I tried looking for a "Test mode" in [Stripe documentation](https://woocommerce.com/document/stripe-tax/), unfortunately they don't seem to offer it. I wasn't able to create a test account since it required a social security number for US account. Perhaps you might be able to use an SG account to test? 😄

File: plugins/woocommerce/package.json (hunk 1 above)
Comment (ilyasfoo) [reply plugins/woocommerce/package.json]: Solved

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php (hunk 7 above)
Comment (rjchow) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php:176]: I think Ukraine would fall under the same category as the *'d ones, as it's "Remote Sales". Not sure why its' not even appearing with an * though. Could be an oversight by Stripe or political concerns

File: plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx (hunk 4 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx]: I think this makes sense!

File: plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx (hunk 5 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx]: I think this makes sense!

File: plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx (hunk 6 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/task-lists/fills/tax/stripe-tax/card.tsx]: Yep it's a new track, I couldn't find anything similar either. Will take your suggestion, thanks!

**Comment by rjchow** [comment]
As discussed with @ilyasfoo I've gone through the full Stripe Tax set up and created follow up tasks in https://github.com/woocommerce/team-ghidorah/issues/355 .

**Review by ilyasfoo — APPROVED** [review]
LGTM! 🚢
